app/controllers/MessagesController.scala

```diff
@@ -69,4 +76,26 @@ class MessagesController @Inject()(
           Future.successful(NotImplemented)
       }
   }
+
+  def getMessage(arrivalId: ArrivalId, messageId: MessageId): Action[AnyContent] = authenticateForRead(arrivalId) {
+    implicit request =>
+      if (request.arrival.messages.isDefinedAt(messageId.index) && request.arrival.messages(messageId.index).optStatus != Some(SubmissionFailed))
+        Ok(Json.toJson(ResponseMovementMessage.build(arrivalId, messageId, request.arrival.messages(messageId.index))))
+      else NotFound
+  }
+
+  def getMessages(arrivalId: ArrivalId): Action[AnyContent] = authenticateForRead(arrivalId) {
+    implicit request =>
+      Ok(
+        Json.toJson(ResponseArrival.build(
```

Comment:
Can we use `Json.toJsObject`? This may force us to define ResponseArrival and ResponseMessages as OWrites